app/components/SheetMusic/comments/Comments.jsx

```diff
@@ -0,0 +1,50 @@
+import fluxMixin from 'flummox/mixin';
+import React from 'react';
+
+import Comment from './Comment';
+import ReplyBox from './ReplyBox';
+
+function retrieveInitialData(flux, id) {
+  let sheetMusicActions = flux.getActions('sheetmusic');
+  return Promise.all([
+    sheetMusicActions.getComments(parseInt(id, 10), flux),
+  ]);
+}
+
+export default React.createClass({
+  propTypes: {
+    /**
+     * The id of the sheet music.
+     */
+    id: React.PropTypes.number.isRequired,
+  },
+
+  mixins: [
+    fluxMixin({
+      comments: store => ({
+        comments: store.state.comments,
+      }),
+    }),
+  ],
+
+  componentDidMount() {
+    retrieveInitialData(this.flux, this.props.id);
```

Comment:
So how I have it implemented right now is that it retrieves the data right after the comments component is rendered.  So sheet music data is retrieved, passes the sheet music id to the comments component, comments component does another api request to grab the list of comments.  Since the comments component may be used in multiple places (maybe?), is it still possible for the server to know what data to retrieve for the component? (no props information)

Or maybe I shouldn't have retrieveInitialData in the comments component in the first place. What if I made the comments api call from the viewer, and I passed in the list of comments object into the comments component?

What do you think?